tools/start_server.py

```diff
@@ -0,0 +1,170 @@
+      rpc_service = self.service_class(_websocket, _path)
+  
+      async for message in _websocket:
+        await rpc_service.on_receive(message)
+  
+  class SampleRPCService():
+    def __init__(self, _websocket, _path):
+      print('Created RPC service')
+      self.websocket = _websocket
+      self.path = _path
+    
+    async def on_receive(self, _msg):
+      print('Message received: %s' % _msg)
+
+
+class RPCService(WSServer.SampleRPCService):
+  def __init__(self, _websocket, _path):
+    super().__init__(_websocket, _path)
+    self.rpc_mapping = {
+      "GetPTSFileContent": self.handle_get_pts_file_content,
+      "SavePTUToFile": self.handle_save_ptu_to_file,
+    }
+
+  async def send(self, _msg):
+    print('***Sending message: %s' % _msg)
+    await self.websocket.send(_msg)
+  
+  async def send_error(self, _error_msg):
+    err = {'success': False, 'info': _error_msg}
+    print(json.dumps(err))
+    await self.send(json.dumps(err))
+
+  async def on_receive(self, _msg):
+    print('***Message received: %s' % _msg)
+    request_message = None
+
+    try:
+      request_message = json.loads(_msg)
+    except ValueError as err:
+      await self.send_error('Invalid JSON received: %s' % err)
+      return
+
+    if 'method' not in request_message:
+      await self.send_error('Received message does not contain manadatory field \'method\'')
+      return
+
+    if 'params' not in request_message:
+      await self.send_error('Received message does not contain manadatory field \'params\'')
+      return
+
+    response_message = self.handle_request(request_message['method'], request_message['params'])
+
+    if response_message != None:
+      await self.send(json.dumps(response_message))
+
+  def handle_request(self, _method_name, _params):
+    # print('***Handling RPC request: {"method": %s, "params": %s}' % (_method, _params))
+    if _method_name not in self.rpc_mapping:
+      return self.gen_error_msg('Method handler for %s was not found! Message ignored' % _method_name)
+
+    return self.rpc_mapping[_method_name](_method_name, _params)
+
+  def handle_get_pts_file_content(self, _method_name, _params):
+    
+    if 'fileName' not in _params:
+      return self.gen_error_msg('Missing mandatory param \'fileName\'')
+
+    file_name = _params["fileName"]
+    file_content = None
+
+    with open(file_name, 'r') as json_file:
+      file_content = json.load(json_file)
+
+    return {
+      "method": _method_name,
+      "success": True,
+      "params":{
+        "content": json.dumps(file_content)
+      }
+    }
+
+  def handle_save_ptu_to_file(self, _method_name, _params):
+    if 'fileName' not in _params:
+      return self.gen_error_msg('Missing mandatory param \'fileName\'')
+    if 'content' not in _params:
+      return self.gen_error_msg('Missing mandatory param \'content\'')
+
+    file_name = _params["fileName"]
+    content = json.loads(_params["content"])
+
+
+    with open(file_name, 'w') as json_file:
```

Comment:
should any security be added here?

Reply:
Could you elaborate on what you mean by security. 

If you are referring to encrypting and decrypting the policy table, I think that step is technically supposed to be handled by the HMI itself https://github.com/smartdevicelink/sdl_evolution/blob/master/assets/proposals/0248-hmi-ptu-support/diagram_ptu_external_proprietary_enhanced.png.

Reply:
I guess I mean in a more general sense, and perhaps we don't have to consider these concerns.

My point was that the ability to write arbitrary JSON to an arbitrary file is pretty powerful. If a malicious actor was in control of the PTU endpoint's response that would give them quite a lot of ability.

The type of security I meant is maybe checking that we are writing to a file in a subdir or not overwriting another file, etc.

Reply:
Added a check in https://github.com/smartdevicelink/generic_hmi/commit/2809eb2d74fd8d68ef46bd38fd52eaebfe76c91b